packages/plans-grid/src/plans-table/index.tsx

```diff
@@ -54,34 +56,39 @@ const PlansTable: React.FunctionComponent< Props > = ( {
 
 	const [ allPlansExpanded, setAllPlansExpanded ] = useState( defaultAllPlansExpanded );
 
+	const getPlanProduct = useSelect( ( select ) => select( PLANS_STORE ).getPlanProduct );
+
 	return (
 		<div className="plans-table">
-			{ supportedPlans.map(
-				( plan ) =>
-					plan && (
-						<PlanItem
-							popularBadgeVariation={ popularBadgeVariation }
-							allPlansExpanded={ allPlansExpanded }
-							key={ plan.periodAgnosticSlug }
-							slug={ plan.periodAgnosticSlug }
-							domain={ currentDomain }
-							tagline={
-								( showTaglines && customTagLines?.[ plan.periodAgnosticSlug ] ) ?? plan.description
-							}
-							CTAVariation={ CTAVariation }
-							features={ plan.features ?? [] }
-							billingPeriod={ billingPeriod }
-							isPopular={ plan.isPopular }
-							isFree={ plan.isFree }
-							name={ plan?.title.toString() }
-							isSelected={ plan.productIds.indexOf( selectedPlanProductId as never ) > -1 }
-							onSelect={ onPlanSelect }
-							onPickDomainClick={ onPickDomainClick }
-							onToggleExpandAll={ () => setAllPlansExpanded( ( expand ) => ! expand ) }
-							disabledLabel={ disabledPlans?.[ plan.periodAgnosticSlug ] }
-						></PlanItem>
-					)
-			) }
+			{ supportedPlans
+				.filter( ( plan ) => !! plan )
+				.map( ( plan ) => (
+					<PlanItem
+						popularBadgeVariation={ popularBadgeVariation }
+						allPlansExpanded={ allPlansExpanded }
+						key={ plan.periodAgnosticSlug }
+						slug={ plan.periodAgnosticSlug }
+						domain={ currentDomain }
+						tagline={
+							( showTaglines && customTagLines?.[ plan.periodAgnosticSlug ] ) ?? plan.description
+						}
+						CTAVariation={ CTAVariation }
+						features={ plan.features ?? [] }
+						billingPeriod={ billingPeriod }
+						isPopular={ plan.isPopular }
+						isFree={ plan.isFree }
+						name={ plan?.title.toString() }
+						isSelected={
+							!! selectedPlanProductId &&
+							selectedPlanProductId ===
+								getPlanProduct( plan.periodAgnosticSlug, billingPeriod )?.productId
+						}
+						onSelect={ onPlanSelect }
+						onPickDomainClick={ onPickDomainClick }
+						onToggleExpandAll={ () => setAllPlansExpanded( ( expand ) => ! expand ) }
+						disabledLabel={ disabledPlans?.[ plan.periodAgnosticSlug ] }
+					></PlanItem>
+				) ) }
```

Comment:
Same as https://github.com/Automattic/wp-calypso/pull/48622/files#r566841545

The diff is because of a change of indentation.
Also I added `.filter( ( plan ) => !! plan )` to avoid doing `plan && (<PlanItem...`